pkg/utils/csvutils/csv_rbac_utils.go

```diff
@@ -0,0 +1,114 @@
+package csvutils
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"unicode"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/types"
+	"github.com/operator-framework/operator-registry/pkg/registry"
+)
+
+const wildCardStr = "*"
+
+// Checks if secrets and configmaps without explicitly defined resource names
+// are accessed at the cluster scope.
+func CheckForConfidentialObjAccessAtClusterScope(csvPermissions *types.CsvPermissions) bool {
+	filterConds := types.RuleFilter{
+		PermissionType: types.ClusterPermissionType,
+		ApiGroupFilterObj: &types.FilterObj{
+			Args:         []string{""},
+			OperatorName: types.InOperator,
+		},
+		ResourcesFilterObj: &types.FilterObj{
+			Args:         []string{"secrets", "configmaps"},
+			OperatorName: types.AnyOperator,
+		},
+		ResourceNamesFilterObj: &types.FilterObj{
+			Args:         []string{},
+			OperatorName: types.DoesNotExistOperator,
+		},
+	}
+	matchedRules := csvPermissions.FilterRules(filterConds)
+	return len(matchedRules) > 0
+}
+
+// Checks if any rules have "*" defined in its apiGroup definition.
+func WildCardApiGroupPresent(csvPermissions *types.CsvPermissions) bool {
+	filterConds := types.RuleFilter{
+		PermissionType: types.AllPermissionType,
+		ApiGroupFilterObj: &types.FilterObj{
+			Args:         []string{wildCardStr},
+			OperatorName: types.InOperator,
+		},
+	}
+	matchedRules := csvPermissions.FilterRules(filterConds)
+	return len(matchedRules) > 0
+}
+
+// Checks if any rules have "*" defined under resources.(For non-operator owned apis.)
+func WildCardResourcePresent(csvPermissions *types.CsvPermissions, ownedApis []string) bool {
+	filterConds := types.RuleFilter{
+		PermissionType: types.AllPermissionType,
+		ApiGroupFilterObj: &types.FilterObj{
+			Args:         ownedApis,
+			OperatorName: types.NotEqualOperator,
+		},
+		ResourcesFilterObj: &types.FilterObj{
+			Args:         []string{wildCardStr},
+			OperatorName: types.InOperator,
+		},
+	}
+	matchedRules := csvPermissions.FilterRules(filterConds)
+	return len(matchedRules) > 0
+}
+
+func GetApisOwned(csv *registry.ClusterServiceVersion) ([]string, error) {
+	ownedApis, _, err := csv.GetCustomResourceDefintions()
+	if err != nil {
+		return nil, err
+	}
+	result := make([]string, 0)
+	for _, api := range ownedApis {
+		if api != nil {
+			result = append(result, trimWhiteSpace(api.Group))
+		}
+	}
+	return result, nil
+}
+
+func GetPermissions(csv *registry.ClusterServiceVersion) (*types.CsvPermissions, error) {
+	var objmap map[string]*json.RawMessage
```

Comment:
To decode the raw json, I feel it would be better to use the actual types from `operator-framework/api`: https://github.com/operator-framework/api/blob/7c97612f258921a973da23fb55dcd77892401be4/pkg/operators/v1alpha1/clusterserviceversion_types.go#L274

An example of this unmarshalling strategy is shown in AM0007: https://github.com/mt-sre/addon-metadata-operator/blob/master/pkg/validators/am0007_csv_install_modes.go#L64

Where you use a custom type.